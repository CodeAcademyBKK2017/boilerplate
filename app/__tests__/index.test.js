import 'react-native';
import App from '../index';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <App />
    );
    expect(tree).toBeDefined();
  });
  //   it('getName: Should return Yo', () => { // example to test class methods
  //     const props = {};
  //     const wrapper = shallow(<App {...props}/>);
  //     const instance = wrapper.instance();
  //     expect(instance.getName()).toEqual('Yo');
  //   });

  it('renders onTypeContent true', () => {
    const wrapper = shallow(<App/>);
    const instance = wrapper.instance();
    instance.onTypeContent('test');
    expect(4).toEqual(instance.state.count);
    expect('test').toEqual(instance.state.inputContent);
  });

  it('renders onTypeContent false', () => {
    const wrapper = shallow(<App/>);
    const instance = wrapper.instance();
    instance.onTypeContent('test');
    expect(2).not.toEqual(instance.state.count);
    expect('test11').not.toEqual(instance.state.inputContent);
  });

  it('renders onTypeTitle true', () => {
    const wrapper = shallow(<App/>);
    const instance = wrapper.instance();
    instance.onTypeTitle('test');
    expect('test').toEqual(instance.state.inputTitle);
  });

  it('renders onSaveNote true', () => {
    const wrapper = shallow(<App/>);
    const instance = wrapper.instance();
    instance.setState({'inputTitle': 't1', 'inputContent': 't2'});
    instance.onSaveNote();
    expect([{'title': 't1', 'content': 't2'}]).toMatchObject(instance.state.note);
  });
});
