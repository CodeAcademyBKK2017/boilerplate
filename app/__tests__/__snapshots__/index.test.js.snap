// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendersItem correctly 1`] = `
<Swipeout
  autoClose={true}
  disabled={false}
  right={
    Array [
      Object {
        "backgroundColor": "lightblue",
        "onPress": [Function],
        "text": "View",
      },
      Object {
        "backgroundColor": "red",
        "onPress": undefined,
        "text": "Delete",
      },
    ]
  }
  rowID={-1}
  sectionID={-1}
  sensitivity={50}
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "900",
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          title
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={undefined}
        >
          content
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</Swipeout>
`;
